src/main/flight/alt_hold.c

```diff
@@ -131,43 +131,42 @@ float altitudePidCalculate(void)
     return output; // motor units, eg 100 means 10% of available throttle 
 }
 
-void simplePidInit(float kp, float ki, float kd, float kf)
+void altHoldPidInit(void)
+{
+    altitudePidCoeffs_t data;
+    getAltitudePidCoeffs(&data);
+    altHoldPid.kp = data.kp;
+    altHoldPid.ki = data.ki;
+    altHoldPid.kd = data.kd;
+    altHoldPid.kf = data.kf;
+    altHoldPid.integral = 0.0f;
+}
+
+void getCurrentAltitudeValues(void)
 {
-    simplePid.kp = kp;
-    simplePid.ki = ki;
-    simplePid.kd = kd;
-    simplePid.kf = kf;
-    simplePid.previousAltitude = 0.0f;
-    simplePid.integral = 0.0f;
+    altitudeData_t data;
+    getAltitudeData(&data);
+    altHoldState.measuredAltitudeCm = data.altitudeCm;
+    altHoldState.altitudeDerivativeCmS = data.altitudeDerivativeCmS;
 }
 
 void altHoldReset(void)
 {
+    altHoldPidInit(); // initialise PID coefficients with values from position_control.c, force iTerm to zero
+    getCurrentAltitudeValues(); // need current altitude to set target altitude
     altHoldState.targetAltitudeCm = altHoldState.measuredAltitudeCm;
```

Comment:
If you don't need the temporary copy to manipulate it, you can get rid of this altogether and use `getAltitude()` directly where it is needed.
```suggestion
    altHoldState.targetAltitudeCm = getAltitude();
```

Reply:
done, throughout